extensions/notification/queueProcessor/QueueProcessor.js

```diff
@@ -141,6 +165,7 @@ class QueueProcessor extends EventEmitter {
      */
     start(options, done) {
         async.series([
+            next => this._setupZookeeper(next),
```

Comment:
can't we setup zookeper inside the `ZookeeperConfigManager` ?

Reply:
The queue populator creates a zookeeper client for all queue populator extensions, we kind of have to do it this way in the queue processor.

Reply:
I don't understand why.
* this specific attribute (`zkClient`) is initialized here, once for every call to `start()`
* it seems it is only used on the next function (`_setupNotificationConfigManager`)
* this function does not store zkClient, it just passes it to notificationManager constructor, which passes it only to the zookeerConfigManager...

--> It seems to me we may as well initialize the zkClient inside `zookeeperConfigManager` constructor (and avoid an extra zkClient in Zenko context)

Reply:
ok, i see what you mean: issue is not here, but in QueuePopulator where we want to reuse the one we already have (even in Zenko).

- However, this current implementation creates a useless zkClient in QueueProcessor on Zenko : which should be avoided.
- One way to avoid this may be to pass both zkConfig and an (optional) zkClient to `NotificationConfigManager` : so that if provided with a client, it can use it (QueuePopulator case) but it will otherwise create one from the config (QueueProcessor case).